src/Nancy/DefaultObjectSerializer.cs

```diff
@@ -16,19 +33,33 @@ public string Serialize(object sourceObject)
         {
             if (sourceObject == null)
             {
-                return String.Empty;
+                return string.Empty;
             }
 
-            var formatter = new BinaryFormatter();
+            dynamic serializedObject = null;
 
-            using (var outputStream = new MemoryStream())
+            //In the case of adding values to cookies it is a string else an object for sessions
+            if (sourceObject is string)
             {
-                formatter.Serialize(outputStream, sourceObject);
-
-                var outputBytes = outputStream.GetBuffer();
+                serializedObject = sourceObject;
+            }
+            else
+            {
+                var sourceType = this.assemblyCatalog.GetAssemblies().Select(assembly => assembly.GetType(sourceObject.GetType().FullName)).FirstOrDefault(type => type != null);
+                if (sourceType == null)
+                {
+                    throw new DllNotFoundException("Unable to find type " + sourceObject.GetType() + " in its assembly to serialize");

```

Comment:
Not sure this exception is the best one. From a user point of view you want to know that it was trying to deserialize something (session state?) and was unable to find the type of the session state data. The exception here could be fine if it is caught higher up and made more user-friendly